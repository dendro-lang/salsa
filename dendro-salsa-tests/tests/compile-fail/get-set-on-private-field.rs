#[salsa::jar(db = Db)]
pub struct Jar(a::MyInput);

mod a {
    use crate::Jar;

    #[salsa::input(jar = Jar)]
    pub struct MyInput {
        field: u32,
    }
}

pub trait Db: salsa::DbWithJar<Jar> {}

#[salsa::db(Jar)]
#[derive(Default)]
struct Database {
    storage: salsa::Storage<Self>,
}

impl salsa::Database for Database {}

impl Db for Database {}

fn main() {
    let mut db = Database::default();
    let input = a::MyInput::new(&mut db, 22);

    input.field(&db);
    input.set_field(&mut db).to(23);
}
