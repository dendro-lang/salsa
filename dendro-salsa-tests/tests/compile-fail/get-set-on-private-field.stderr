error[E0624]: method `field` is private
  --> tests/compile-fail/get-set-on-private-field.rs:29:11
   |
7  |     #[salsa::input(jar = Jar)]
   |     -------------------------- private method defined here
...
29 |     input.field(&db);
   |           ^^^^^ private method

error[E0624]: method `set_field` is private
  --> tests/compile-fail/get-set-on-private-field.rs:30:11
   |
7  |     #[salsa::input(jar = Jar)]
   |     -------------------------- private method defined here
...
30 |     input.set_field(&mut db).to(23);
   |           ^^^^^^^^^ private method
